napari/_qt/qt_main_window.py

```diff
@@ -105,22 +101,31 @@ def __init__(self, viewer, *, show: bool = True):
 
             # Will patch based on config file.
             perf_config.patch_callables()
-
         if (
             platform.system() == "Windows"
             and not getattr(sys, 'frozen', False)
-            and self._napari_app_id
+            and hasattr(viewer, "_napari_app_id")
+            and viewer._napari_app_id
         ):
             import ctypes
 
             ctypes.windll.shell32.SetCurrentProcessExplicitAppUserModelID(
-                self._napari_app_id
+                viewer._napari_app_id
             )
 
         logopath = os.path.join(
             os.path.dirname(__file__), '..', 'resources', 'logo.png'
         )
-        app.setWindowIcon(QIcon(logopath))
+
+        if (
+            hasattr(viewer, "_napari_global_logo")
+            and viewer._napari_global_logo
```

Comment:
alternatively `getattr(viewer, "_napari_global_logo", True)`

Reply:
that's better